src/rgw/rgw_rados.cc

```diff
@@ -2847,10 +2847,13 @@ class RGWRadosThread {
 
   atomic_t down_flag;
 
+  string thread_name;
+
   virtual uint64_t interval_msec() = 0;
   virtual void stop_process() {}
 public:
-  RGWRadosThread(RGWRados *_store) : worker(NULL), cct(_store->ctx()), store(_store) {}
+  RGWRadosThread(RGWRados *_store, string thread_name = "radosgw") 
```

Comment:
pass the string by const ref to avoid an extra copy, `const string& thread_name = "radosgw"`

(same with the `RGWSyncProcessorThread` constructor)